file.go

```diff
@@ -541,9 +541,6 @@ func (f *filePages) readDictionaryPage(header *format.PageHeader, page *dataPage
 	if header.DictionaryPageHeader == nil {
 		return ErrMissingPageHeader
 	}
-	if f.index > 0 {
-		return ErrUnexpectedDictionaryPage
-	}
```

Comment:
On second thought, the following appears to be an appropriate spot for the index check:
https://github.com/segmentio/parquet-go/blob/86cb68738df079b236e082d0ee49436b86981470/file.go#L482-L486

ie.. if we encounter a dictionary page beyond the first index while scanning, that should result in a `ErrUnexpectedDictionaryPage`. However, if we are lazy-loading a dictionary page from the top of the columnchunk, the index of the current page doesn't matter.

I'm not sure how I can come up with a test for this though, I can move the check to the new location if it looks ok to you?